lib/livebook/migration.ex

```diff
@@ -1,17 +1,38 @@
 defmodule Livebook.Migration do
-  @doc """
-  Runs all migrations.
-  """
-  @spec migrate() :: :ok
-  def migrate() do
-    delete_local_host_hub()
+  use GenServer, restart: :temporary
+
+  # We version our storage so we can remove migrations in the future.
+  # We also documented tables and IDs used in previous versions,
+  # as those must be avoided in the future.
+  #
+  # ## v1 (Livebook v0.11)
+  #
+  # * Deleted hubs.local-host
+  # * Migrated secrets to hub_secrets
+  # * Migrated filesystems to file_systems
+  # * Migrated settings.global.default_file_system_id to settings.global.default_dir
+  #
+  @migration_version 1
+
+  def start_link(_opts) do
+    GenServer.start_link(__MODULE__, :ok)
+  end
+
+  def init(:ok) do
     insert_personal_hub()
-    move_app_secrets_to_personal_hub()
     add_personal_hub_secret_key()
```

Comment:
Should this also be a part of v1? In the feature if we drop the whole storage, then `insert_personal_hub` will create a valid hub.